### PR TITLE
metrics.publish()

### PR DESCRIPTION
**Pull request by eddyashton**

### CODE REVIEW

File: tests/upload_pico_metrics.py [hunk 1]
```diff
@@ -84,3 +84,4 @@
             upload_name = f"{name}_{i}" if many_results else name
             LOG.debug(f"Uploading metric: {upload_name} = {result}")
             metrics.put(upload_name, result)
+    metrics.publish()
```
Comment (achamayou) [inline tests/upload_pico_metrics.py]: We should probably turn Metrics into a context manager to make it slightly less likely what one can forget to publish (what are the changes you want to add metrics but not publish them).

**Comment by jumaffre** [comment]
Please do not merge this before https://github.com/jumaffre/cimetrics/issues/31 is resolved (by the end of today, hopefully).

File: tests/upload_pico_metrics.py (hunk 1 above)
Comment (achamayou) [reply tests/upload_pico_metrics.py]: https://github.com/jumaffre/cimetrics/issues/41

**Comment by cimetrics** [comment]
images

**Comment by jumaffre** [comment]
> Please do not merge this before [jumaffre/cimetrics#31](https://github.com/jumaffre/cimetrics/issues/31) is resolved (by the end of today, hopefully).

Looks like it now works. A known limitation is the fact that the metrics value won't be displayed the first time around (but will be once this is merge to `master`). We should probably start thinking about filtering metrics to be displayed too.

**Pull request merged**